src/meta/ClientBasedGflagsManager.cpp

```diff
@@ -0,0 +1,142 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "meta/ClientBasedGflagsManager.h"
+
+DECLARE_int32(load_config_interval_secs);
+
+namespace nebula {
+namespace meta {
+
+ClientBasedGflagsManager* ClientBasedGflagsManager::instance(MetaClient* client) {
+    static auto gflagsMan = std::unique_ptr<ClientBasedGflagsManager>(
+            new ClientBasedGflagsManager(client));
+    static std::once_flag initFlag;
+    std::call_once(initFlag, &ClientBasedGflagsManager::init, gflagsMan.get());
+    return gflagsMan.get();
+}
+
+ClientBasedGflagsManager::ClientBasedGflagsManager(MetaClient *client) {
+    metaClient_ = client;
+    CHECK_NOTNULL(metaClient_);
+}
+
+ClientBasedGflagsManager::~ClientBasedGflagsManager() {
+    if (nullptr != metaClient_) {
+        metaClient_ = nullptr;
+    }
+}
+
+void ClientBasedGflagsManager::init() {
+    CHECK(bgThread_.start("ClientBasedGflagsManager_bgThread"));
+    declareGflags();
+    regCfgThreadFunc();
+    loadCfgThreadFunc();
+}
+
+template<typename ValueType>
+folly::Future<StatusOr<bool>> ClientBasedGflagsManager::set(const cpp2::ConfigModule& module,
+                                                            const std::string& name,
+                                                            const cpp2::ConfigType& type,
+                                                            const ValueType& value) {
+    std::string valueStr;
+    valueStr.append(reinterpret_cast<const char*>(&value), sizeof(value));
+    return metaClient_->setConfig(module, name, type, valueStr);
+}
+
+template<>
```

Comment:
Do we need the template keyword?

Reply:
It's template specialization for std::string